libs/pika/algorithms/tests/unit/algorithms/CMakeLists.txt

```diff
@@ -147,10 +147,18 @@ set(tests
     uninitialized_moven
     uninitialized_value_construct
     uninitialized_value_constructn
-    unique
     unique_copy
 )
 
+# Cray's clang compiler produces segfaults in release mode on the unique test.
+# This seems to be a compiler bug. We disable the test for all non-Debug builds
+# with clang.
+if(NOT (NOT "${CMAKE_BUILD_TYPE}" STREQUAL "Debug" AND "${CMAKE_COMPILER_ID}"
+                                                       STREQUAL "Clang")
+)
+  list(APPEND tests unique)
+endif()
+
```

Comment:
This problem is listed under `unique` on https://github.com/pika-org/pika/issues/191.